bundles/org.openhab.binding.gpio/src/main/java/org/openhab/binding/gpio/internal/handler/PigpioRemoteHandler.java

```diff
@@ -57,60 +63,368 @@ public class PigpioRemoteHandler extends BaseThingHandler {
+                    "Cannot connect to PiGPIO Service on remote raspberry. Invalid Port.");
             return;
         }
-        thing.getChannels().forEach(channel -> {
+
+        createChannelHandlers();
+
+        logger.debug("gpio binding initialized");
+
+        connectionJob = scheduler.submit(() -> {
+            connectionPollWorker();
+        });
+    }
+
+    protected void clearChannelHandlers() {
+        for (ChannelHandler handler : channelHandlers.values()) {
+            handler.dispose();
+        }
+        channelHandlers.clear();
+    }
+
+    protected void createChannelHandlers() {
+        clearChannelHandlers();
+        this.getThing().getChannels().forEach(channel -> {
             ChannelUID channelUID = channel.getUID();
             ChannelTypeUID type = channel.getChannelTypeUID();
+
             try {
                 if (CHANNEL_TYPE_DIGITAL_INPUT.equals(type)) {
                     GPIOInputConfiguration configuration = channel.getConfiguration().as(GPIOInputConfiguration.class);
-                    channelHandlers.put(channelUID, new PigpioDigitalInputHandler(configuration, jPigpio, scheduler,
+                    this.channelHandlers.put(channelUID, new PigpioDigitalInputHandler(configuration, scheduler,
                             state -> updateState(channelUID.getId(), state)));
                 } else if (CHANNEL_TYPE_DIGITAL_OUTPUT.equals(type)) {
                     GPIOOutputConfiguration configuration = channel.getConfiguration()
                             .as(GPIOOutputConfiguration.class);
-                    channelHandlers.put(channelUID, new PigpioDigitalOutputHandler(configuration, jPigpio,
-                            state -> updateState(channelUID.getId(), state)));
+                    PigpioDigitalOutputHandler handler = new PigpioDigitalOutputHandler(configuration, scheduler,
+                            state -> updateState(channelUID.getId(), state));
+                    this.channelHandlers.put(channelUID, handler);
                 }
             } catch (PigpioException e) {
-                logger.warn("Failed to initialize {}: {}", channelUID, e.getMessage());
-            } catch (InvalidPullUpDownException e) {
-                logger.warn("Failed to initialize {}: Invalid Pull Up/Down resistor configuration", channelUID);
-            } catch (NoGpioIdException e) {
-                logger.warn("Failed to initialize {}: GpioId is not set", channelUID);
+                logger.warn("Failed to initialize channel {} {}", channelUID, e.toString());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                        String.format("Failed to initialize channel {} {}", channelUID, e.getLocalizedMessage()));
+            } catch (ChannelConfigurationException e) {
+                logger.error("Invalid configuration for channel {} {}", channelUID, e.toString());
```

Comment:
As the thing status including details is allready set, this logger should be removed.